pkg/util/yaml.go

```diff
@@ -0,0 +1,120 @@
+package util
```

Comment:
Should we add unit tests to cover the new functionality introduced by this yaml util?

Reply:
Nah, not strictly necessary, the existing tests cover this. Though at some point we'd want to make smaller unit tests and test these directly.